packages/ia-components/components/audio-players/archive-audio-jwplayer-wrapper.jsx

```diff
@@ -74,44 +105,118 @@ class ArchiveAudioPlayer extends Component {
   }
 
   /**
-   * Check if track index has changed. If so, then play that track
+   * We track when:
+   * - Play8 is loaded
+   * - track is playing
+   * - autoplay is happening
+   * - to change next track
+   *
+   * @inheritdoc
    */
-  componentDidUpdate(prevProps) {
-    const { sourceData: { index = null } } = this.props;
+  componentDidUpdate(prevProps, prevState) {
+    const { sourceData: { index: incomingTrackNum = null } } = this.props;
+    const {
+      playerReady, playlistLoadCount, playerEverStarted, jwplayerInstance, trackNumber, trackStarting
+    } = this.state;
     const { sourceData: prevSourceData } = prevProps;
     const { index: prevIndex } = prevSourceData;
-    const { jwplayerInstance, playerPlaylistIndex } = this.state;
-    const indexIsNumber = Number.isInteger(index);
+    const trackN = parseInt(incomingTrackNum, 10);
+    const isNowMainPlayer = !prevSourceData.hasOwnProperty('index') && trackN;
+    const unsupportedAlbumView = incomingTrackNum === 0; // 0 = album for 3d party players
 
-    if (!indexIsNumber) return;
+    if (!jwplayerInstance
+      || unsupportedAlbumView
+      || Number.isNaN(trackN)
+      || (trackN < 0)
+      || isNowMainPlayer
+    ) {
+      return;
+    }
 
-    const isNowMainPlayer = !prevSourceData.hasOwnProperty('index') && indexIsNumber;
+    const iaPlayerisReady = playlistLoadCount === this.maxPlaylistLoadsUntilPlayerIsReady;
+    if (!playerReady && iaPlayerisReady) {
+      /** IA player starting, let's tell main app where we are starting */
+      this.updateAndPlayTrack({
+        playerReady: true,
+        trackNumber: incomingTrackNum,
+      });
+      return;
+    }
 
-    if (isNowMainPlayer) {
-      // do not autoplay this track any longer
+    if (!prevState.playerReady && playerReady) {
+      /** Play8 w/ JWP is now ready & at rest, we can also rest */
       return;
     }
-    const incomingTrackChange = index !== prevIndex;
 
-    const isOnSameTrack = playerPlaylistIndex === index;
     const playerStatus = jwplayerInstance.getState();
-    const playCurrentTrack = isOnSameTrack && (playerStatus === 'idle');
-    const playTrack = indexIsNumber && (playCurrentTrack || incomingTrackChange);
-    if (playTrack) {
-      this.playTrack({ playerPlaylistIndex: index - 1 || 0 });
+    const incomingTrackChange = incomingTrackNum > prevIndex || (trackNumber !== incomingTrackNum);
```

Comment:
SUGGESTION: Parentheses not necessary here. Or if you really want them, put around both sides.